src/rgw/rgw_op.h

```diff
@@ -57,12 +57,14 @@
 
 #include "include/ceph_assert.h"
 
+#include "rgw_tracer.h"
 using ceph::crypto::SHA1;
 
 struct req_state;
 class RGWOp;
 class RGWRados;
 
+extern thread_local std::unique_ptr<tracing::Tracer> rgw_tracer;
```

Comment:
what's the motivation for thread-local here?

p.s. this variable is declared extern in rgw_op.h and defined in rgw_frontend.cc. these probably belong in rgw_tracer.h and rgw_tracer.cc instead

Reply:
> what's the motivation for thread-local here?
> 

to prevent contention on the lock inside the tracer.

> p.s. this variable is declared extern in rgw_op.h and defined in rgw_frontend.cc. these probably belong in rgw_tracer.h and rgw_tracer.cc instead

agree. the tracer could be used in other contexts than ops/frontend

Reply:
`rgw_tracer` was moved to rgw_tracer.cc, in ffd1262

Reply:
can you point me at the locks we're concerned about?